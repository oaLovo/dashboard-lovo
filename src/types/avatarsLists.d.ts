export type ListDataProps = {
  avatar?: string
  name: string
  subtitle: string
}

export interface AvatarsListsProps {
  listData: ListDataProps[]
}
