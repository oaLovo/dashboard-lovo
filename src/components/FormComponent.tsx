import styled from 'styled-components'
import { FormComponentProps } from '@/types'
import { StyledButton, StyledInput, StyledP } from '@/components'
import { pxToRem } from '@/utils'

const StyledForm = styled.form`
  display: flex;
  flex-direction: column;
  row-gap: ${pxToRem(16)};
`

function FormComponent(props: FormComponentProps) {
  const { inputs, buttons, message } = props
  return (
    <StyledForm>
      {inputs.map((inputProps, index) => (
        <StyledInput key={index} {...inputProps} />
      ))}
      {buttons.map((buttonProps, index) => (
        <StyledButton key={index} {...buttonProps} />
      ))}
      {message && (
        <StyledP className={message.type === 'error' ? 'error' : 'success'}>
          {message.msg}
        </StyledP>
      )}
    </StyledForm>
  )
}

export default FormComponent
