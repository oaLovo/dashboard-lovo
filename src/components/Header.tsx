import styled from 'styled-components'
import { Link } from 'react-router-dom'

import { Logo } from '@/components'

// MUI
import Avatar from '@mui/material/Avatar'
import Box from '@mui/material/Box'
import Container from '@mui/material/Container'

// UTILS
import { pxToRem } from '@/utils'

const StyledHeader = styled.header`
  background-color: ${(props) => props.theme.appBackground};
  border-bottom: ${pxToRem(1)} solid ${(props) => props.theme.appDefaultStroke};
  margin-bottom: ${pxToRem(37)};
  width: 100%;
`

function Header() {
  return (
    <StyledHeader>
      <Container maxWidth="lg">
        <Box
          sx={{
            alignItems: 'center',
            display: 'flex',
            justifyContent: 'space-between',
            height: pxToRem(64),
          }}
        >
          <Link to="/home">
            <Logo height={30} width={73} />
          </Link>
          <Link to="/perfil">
            <Avatar
              alt="DNC Avatar"
              src="/dnc-avatar.jpg"
              sx={{ width: pxToRem(40), height: pxToRem(40) }}
            />
          </Link>
        </Box>
      </Container>
    </StyledHeader>
  )
}

export default Header
