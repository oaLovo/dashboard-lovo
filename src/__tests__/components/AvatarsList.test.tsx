import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import { BrowserRouter as Router } from 'react-router-dom'
import { AvatarsList } from '@/components'
import { themesList } from '@/resources/themesList'
import { Theme } from '@/types'

const mockListData = [
  {
    avatar: '/dnc-avatar.jpg',
    name: 'Nome Sobrenome 1',
    subtitle: 'R$ 1.000,00',
  },
  {
    avatar: '/dnc-avatar.jpg',
    name: 'Nome Sobrenome 2',
    subtitle: 'R$ 1.000,00',
  },
  {
    avatar: '/dnc-avatar.jpg',
    name: 'Nome Sobrenome 3',
    subtitle: 'R$ 1.000,00',
  },
]

describe('AvatarsList Component', () => {
  const renderComponent = (theme: Theme) =>
    render(
      <ThemeProvider theme={theme}>
        <Router>
          <AvatarsList listData={mockListData} />
        </Router>
      </ThemeProvider>
    )

  themesList.forEach(({ name, theme }) => {
    describe(`${name}`, () => {
      it('should match the snapshot', () => {
        const { container } = renderComponent(theme)
        expect(container).toMatchSnapshot()
      })
    })
  })
})
