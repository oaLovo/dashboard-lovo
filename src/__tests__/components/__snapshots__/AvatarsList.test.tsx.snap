// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarsList Component Dark Theme should match the snapshot 1`] = `
.c0 {
  color: inherit;
  font-size: 1rem;
  font-weight: 600;
  line-height: 1.5rem;
}

.c1 {
  color: #89A7CE;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5rem;
}

<div>
  <div
    class="MuiBox-root css-1shl023"
  >
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circular css-1bt7igf-MuiAvatar-root"
      >
        <img
          alt="Nome Sobrenome 1"
          class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
          src="/dnc-avatar.jpg"
        />
      </div>
    </div>
    <div
      class="MuiBox-root css-0"
    >
      <h2
        class="c0"
      >
        Nome Sobrenome 1
      </h2>
      <span
        class="c1"
      >
        R$ 1.000,00
      </span>
    </div>
  </div>
  <div
    class="MuiBox-root css-1shl023"
  >
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circular css-1bt7igf-MuiAvatar-root"
      >
        <img
          alt="Nome Sobrenome 2"
          class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
          src="/dnc-avatar.jpg"
        />
      </div>
    </div>
    <div
      class="MuiBox-root css-0"
    >
      <h2
        class="c0"
      >
        Nome Sobrenome 2
      </h2>
      <span
        class="c1"
      >
        R$ 1.000,00
      </span>
    </div>
  </div>
  <div
    class="MuiBox-root css-1shl023"
  >
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circular css-1bt7igf-MuiAvatar-root"
      >
        <img
          alt="Nome Sobrenome 3"
          class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
          src="/dnc-avatar.jpg"
        />
      </div>
    </div>
    <div
      class="MuiBox-root css-0"
    >
      <h2
        class="c0"
      >
        Nome Sobrenome 3
      </h2>
      <span
        class="c1"
      >
        R$ 1.000,00
      </span>
    </div>
  </div>
</div>
`;

exports[`AvatarsList Component Light Theme should match the snapshot 1`] = `
.c0 {
  color: inherit;
  font-size: 1rem;
  font-weight: 600;
  line-height: 1.5rem;
}

.c1 {
  color: #828282;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5rem;
}

<div>
  <div
    class="MuiBox-root css-1shl023"
  >
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circular css-1bt7igf-MuiAvatar-root"
      >
        <img
          alt="Nome Sobrenome 1"
          class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
          src="/dnc-avatar.jpg"
        />
      </div>
    </div>
    <div
      class="MuiBox-root css-0"
    >
      <h2
        class="c0"
      >
        Nome Sobrenome 1
      </h2>
      <span
        class="c1"
      >
        R$ 1.000,00
      </span>
    </div>
  </div>
  <div
    class="MuiBox-root css-1shl023"
  >
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circular css-1bt7igf-MuiAvatar-root"
      >
        <img
          alt="Nome Sobrenome 2"
          class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
          src="/dnc-avatar.jpg"
        />
      </div>
    </div>
    <div
      class="MuiBox-root css-0"
    >
      <h2
        class="c0"
      >
        Nome Sobrenome 2
      </h2>
      <span
        class="c1"
      >
        R$ 1.000,00
      </span>
    </div>
  </div>
  <div
    class="MuiBox-root css-1shl023"
  >
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circular css-1bt7igf-MuiAvatar-root"
      >
        <img
          alt="Nome Sobrenome 3"
          class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
          src="/dnc-avatar.jpg"
        />
      </div>
    </div>
    <div
      class="MuiBox-root css-0"
    >
      <h2
        class="c0"
      >
        Nome Sobrenome 3
      </h2>
      <span
        class="c1"
      >
        R$ 1.000,00
      </span>
    </div>
  </div>
</div>
`;
